### PR TITLE
Features revision

### PR DESCRIPTION
**Pull request by ronniebermejo**

### CODE REVIEW

File: app/channels/product_channel.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Be sure to restart your server when you modify this file. Action Cable runs in a loop that does not support auto reloading.
+class ProductChannel < ApplicationCable::Channel
+
+  def subscribed
+    # stream_from "some_channel"
+    # stream_from "product_channel"
```
Comment (ronniebermejo) [inline app/channels/product_channel.rb]: ?

File: app/mailers/order_mailer.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class OrderMailer < ApplicationMailer
+  default from: 'admin@greenjocote.com'
+  def purchased(user, total)
+    @total = total
+    mail(to: user.email, subject: "Thank you for your purchase - order #{}")
```
Comment (ronniebermejo) [inline app/mailers/order_mailer.rb]: `"Thank you for your purchase - order #{}"`, 
it seems there is  something missing inside the `#{}`

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -45,15 +45,16 @@
             <li>
               <%= link_to "Contact", static_pages_contact_path %>
             </li>
+            <%end%>
             <li>
-              <% if user_signed_in? %>
+              <% if signed_in? %>
```
Comment (ronniebermejo) [inline app/views/layouts/application.html.erb]: 👍

File: app/views/products/show.html.erb [hunk 4]
```diff
@@ -41,29 +41,34 @@
         <strong>Colour:</strong>
         <%= @product.colour %>
       </p>
-
+<% end %>
```
Comment (ronniebermejo) [inline app/views/products/show.html.erb]: not sure where the corresponding `if` is, something strange is going on here 🌵

File: spec/controllers/products_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    @product = FactoryGirl.create(:product)
+  end
+
+  context 'GET #index' do
+    before do
+      get :index
+    end
+
+    it 'responds successfully with an HTTP 200 status code' do
+      expect(response).to be_success
```
Comment (ronniebermejo) [inline spec/controllers/products_controller_spec.rb]: This seems a duplicated test.

File: spec/controllers/products_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    end
+
+    it 'renders the index template' do
+      expect(response).to render_template('index')
+    end
+  end
+
+  context 'GET #show' do
+    before do
+      get :show, id: @product.id
```
Comment (ronniebermejo) [inline spec/controllers/products_controller_spec.rb]: Read this great resource: http://betterspecs.org/

File: spec/controllers/products_controller_spec.rb [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    end
+
+    it 'responds successfully with an HTTP 200 status code' do
+      expect(response).to be_success
+      expect(response).to have_http_status(200)
```
Comment (ronniebermejo) [inline spec/controllers/products_controller_spec.rb]: Single expectation test /

File: spec/controllers/products_controller_spec.rb [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+    end
+
+    it 'renders the index template' do
+      expect(response).to render_template('index')
+    end
+  end
+
+  context 'GET #show' do
+    before do
```
Comment (ronniebermejo) [inline spec/controllers/products_controller_spec.rb]: Use let and let!

File: spec/controllers/products_controller_spec.rb [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+      end
+      it 'renders the edit template' do
+        expect(response).to render_template('edit')
+      end
+    end
+
+    context 'when user is not logged in' do
+      it 'redirects to login' do
+        get :edit, id: @product.id
+        expect(response).to redirect_to('/login')
```
Comment (ronniebermejo) [inline spec/controllers/products_controller_spec.rb]: 👍

File: spec/factories/product_factory.rb [hunk 10]
```diff
@@ -0,0 +1,9 @@
+FactoryGirl.define do
+  factory :product do
+    name 'Incredible bike'
+    description 'But supposing the invader of domestic bliss to betake himself away at the first rush of the harem\'s lord, then is it very diverting to watch that lord. Gently he insinuates his vast bulk among them again and revels there awhile, still in tantalizing vicinity to young Lothario, like pious Solomon devoutly worshipping among his thousand concubines. Granting other whales to be in sight, the fishermen will seldom give chase to one of these Grand Turks; for these Grand Turks are too lavish of their strength, and hence their unctuousness is small. As for the sons and the daughters they beget, why, those sons and'
+    image_url '#'
+    colour 'red'
```
Comment (ronniebermejo) [inline spec/factories/product_factory.rb]: 👍

File: app/views/products/show.html.erb (hunk 4 above)
Comment (kervin5) [inline app/views/products/show.html.erb]: this is the closing tag for the cache block that is located on the first line.

**Pull request merged**